### PR TITLE
os/board/rtl8730e: Add RTK Key Derive Function API (PBKDF2)

### PR DESCRIPTION
**Pull request by zhongnuo-tang**

### CODE REVIEW

File: os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_common/ameba_ota.c [hunk 1]
```diff
@@ -430,4 +430,28 @@ int OTA_KernelImageSignatureCheck(uint32_t input_addr)
 	return TRUE;
 }
 
+int Ameba_KeyDeriveFunc(const unsigned char *password, size_t plen,
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_common/ameba_ota.c:433]: Could you leave any description about return value?

Comment (zhongnuo-tang) [reply os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_common/ameba_ota.c:451]: Dear @sunghan-chang ,
I have added description for this API.

**Review by sunghan-chang — APPROVED** [review]

**Review by jylee9613 — COMMENTED** [review]
Is it okay not to update this API to the TZ?

**Comment by ZhenBei-Sin** [comment]
> Is KDF API already in Trustzone? There's no update in Trustzone.

Hi @jylee9613 , The rtl8730e Ecode was included in the KDF feature in SE.
commit ID: f935639813b81b471882faba602ea7e6b870c2c1.

After confirming Samsung needed this KDF feature,  we submitted the PR for adding the SMCCC wrapper entry point in the TizenRT branch.

Thank You.

**Pull request merged**